### ISSUE TITLE
APIServer health-check timed out

### ISSUE THREAD

**Issue by dza89** [k0sproject/k0s#1873]
### Before creating an issue, make sure you've checked the following:

- [X] You are running the latest released version of k0s
- [X] Make sure you've searched for existing issues, both open and closed
- [X] Make sure you've searched for PRs too, a fix might've been merged already
- [X] You're looking at docs for the released version, "main" branch docs are usually ahead of released versions.

### Version

v1.23.7+k0s.0

### Platform

```shell
LSB Version:    n/a
Distributor ID: Arch
Description:    Arch Linux
Release:        rolling
Codename:       n/a
```

### What happened?

Can't start k0s with the following command:

`
/usr/local/bin/k0s controller --config=/etc/k0s/k0s.yaml --enable-worker=true
`

### Steps to reproduce

Actually tried it with first k0sctl. 
Did a k0sctl reset, and tried again on the node, with the same error.

1.  Run /usr/local/bin/k0s controller --config=/etc/k0s/k0s.yaml --enable-worker=true
2.  See attachment

### Expected behavior

Run k0s succesfully

### Actual behavior

k0s shuts down

### Screenshots and logs

[k0scontroller.logs.gz](https://github.com/k0sproject/k0s/files/8936159/k0scontroller.logs.gz)

**Comment by jnummelin** [comment]
I was able to reproduce this on AWS Arch AMI.

So there's two issues I needed to fix:
- `/etc/hosts` did not have the usual entry for localhost. Added the line manually
- `/etc/resolv.conf` was empty. Fixed by `ln -rsf /run/systemd/resolve/stub-resolv.conf /etc/resolv.conf` so that it's managed by systemd-resolved.

Due to the static linking of k0s (and all the dependencies too) this results in Golang doing resolving only via the two above mentioned files. As there's systemd-resolved in place on arch, I wonder why does it not do the usual linking for `/etc/resolv.conf` out-of-box.

However, I see three actionable things here for k0s:
- [ ] Document the "assumptions" static linking creates for DNS setup
- [ ] Fix the log level of the api health check failing. currently it's on debug level:
```
Jun 20 10:35:16 ip-172-31-44-57 k0s[935]: time="2022-06-20 10:35:16" level=debug msg="health-check: APIServer not yet healthy: Get \"https://localhost:6443/readyz?verbose\": dial tcp: lookup localhost: Try again"
```
- [ ] Add checking of `/etc/resolv.conf` to our preflight-checks, maybe by trying to resolve e.g. `localhost`.
